base_layer/core/src/mining/miner.rs

```diff
@@ -281,6 +318,25 @@ impl Miner {
                         _ => {wait_for_miner = true;},
                     }
                 },
+                instruction = mining_instruction_event.select_next_some() => {
+                    use MinerInstruction::*;
+                    match instruction.unwrap_or_else(|_| (IgnoreInstruction)) {
+                        PauseMining => {
+                            info!(target: LOG_TARGET, "Mining pause event received from the CLI");
+                            stop_mining_flag.store(true, Ordering::Relaxed);
+                            mining_enabled_by_user.store(false, Ordering::Relaxed);
+                            start_mining = false;
+                            wait_for_miner = true;
```

Comment:
If this is true, an event van turn the miner on again. 
Imagine this:
1: user sets pause.
2: miner stops
3: node goes to lagging state. 
4: node goes to BlocksSynchronized state
5: node starts mining again. 

But just setting this to false means the miner wont be able to listen again here.

Reply:
Nevermind here, missed the ! on line 291

Reply:
I understand the confusion, as it is really difficult to get a clear meaning here.
- Propose to change `start_mining` to `spawn_mining_task`
- Propose to change `let mut wait_for_miner = false;` and `while !wait_for_miner {` to positive logic
- Propose to change `wait_for_miner ` to `wait_for_mining_event`
`